torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,253 @@
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data"""
+import abc
+import glob
+import hashlib
+import os
+from functools import lru_cache
+from typing import Any, Callable, Dict, List, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import NonGeoDataset, RasterDataset
+from .landsat import Landsat7
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class Landsat7Irish(RasterDataset):
+    
+    """ Landsat 7 Cloud Cover Assessment Validation Data.
+
+    The `Landsat 7 Cloud Cover Assessment Validation Data <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
+    Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes, displayed in the biomes listed below. Manually generated cloud masks are used to train and validate cloud cover 
+    assessment algorithms, which in turn are intended to compute the percentage of cloud cover in each scene. *Originally, this was reported as 207 masks, but one mask 
+    was erroneously counted.
+
+    Dataset format:
+    * 102 scenes from this dataset were used in a cloud validation study described in Foga, Scaramuzza et al. (2017)
+    * labels are values [0, 64, 128, 192, 255]
+
+     Dataset classes:
+     * Fill
+     * Cloud Shadow
+     * Clear
+     * Thin Cloud
+     * Cloud
+
+     If you use this dataset in your research, please cite the following papers:
+
+     * U.S. Geological Survey, 2016. L7 Irish Cloud Validation Masks. U.S. Geological Survey data release. doi:10.5066/F7XD0ZWC.
+     * Scaramuzza, P.L., Bouchard, M.A. & Dwyer, J.L. (2012). Development of the Landsat data continuity mission cloud-cover assessment algorithms. IEEE Transactions on Geoscience and Remote Sensing, 50(4), 1140-1154. doi:10.1109/TGRS.2011.2164087.
+     """
+
+    url = "https://huggingface.co/datasets/torchgeo/l7irish"
+    
+    md5 = ["dbb6b5628f50861b9b89f548d25a925f",
+           "cecc72de09aacde4c4f8d7f0cf0d3f6f",
+           "0f8382ca6554fb7cf9aff42226a14f9d",
+           "b17cf6d023f752c533211fdb742f296b",
+           "73923dcaf1b9b79bad82de1aa0740d1e",
+           "3bc9f4c6f8955b10b4d55d23e0ab2da7",
+           "f8f039970256902e6e9ebd6747589294",
+           "8346d73a983396c5d41b577c3a94bc26",
+           "abe19b22b5d031e6b609cc7207706c3d"]
+    
+    biomes = ["austral","boreal","mid_latitude_north","mid_latitude_south",
+              "polar_north","polar_south","subtropical_north",
+              "subtropical_south","tropical"]
+    
+    classes = ["Fill", "Cloud Shadow", "Clear", "Thin Cloud", "Cloud"]
+    
+    cmap = {
+        0: (0,0,0),
+        64: (64,64,64),
+        128: (128,128,128),
+        192: (192,192,192),
+        255: (255,255,255),
+    }
+
```

Comment:
You're missing filename_glob and filename_regex. Without those, this dataset won't actually locate any files. See https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html.

Reply:
Still missing filename_glob